test/runtime/ct.go

```diff
@@ -0,0 +1,433 @@
+package RuntimeTest
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	serverImage = "httpd"
+	ctCleanUpNC = "ct-clean-up-nc.py"
+	ctL4Policy  = "CT-l4-policy.json"
+	netcat      = "netcat"
+)
+
+type connTest struct {
+	src         map[string]string
+	destination map[string]string
+	kind        string
+	mode        string
+}
+
+func (c connTest) String() string {
+	return fmt.Sprintf("%s-%s-%s", c.src["Name"], c.destination["Name"], c.kind)
+}
+
+var _ = Describe("RuntimeConntrackTable", func() {
+
+	var initialized bool
+	var logger *logrus.Entry
+	var vm *helpers.SSHMeta
+	var (
+		HTTPPrivate   = "private"
+		HTTPPublic    = "public"
+		HTTPDummy     = "dummy"
+		netcatPrivate = "ncPrivate"
+		netcatPublic  = "ncPublic"
+		server        = "server"
+		server2       = "server-2"
+		server3       = "server-3"
+		client2       = "client-2"
+		client        = "client"
+	)
+
+	initialize := func() {
+		if initialized == true {
+			return
+		}
+		logger = log.WithFields(logrus.Fields{"testName": "RuntimeConntrack"})
+		logger.Info("Starting")
+		vm = helpers.CreateNewRuntimeHelper(helpers.Runtime, logger)
+		vm.WaitUntilReady(100)
+		vm.NetworkCreate(helpers.CiliumDockerNetwork, "")
+
+		res := vm.SetPolicyEnforcement(helpers.PolicyEnforcementDefault)
+		res.ExpectSuccess()
+
+		res = vm.PolicyDelAll()
+		res.ExpectSuccess()
+
+		initialized = true
+	}
+
+	containersNames := []string{server, server2, server3, client, client2, "netcat"}
+
+	containers := func(mode string) {
+		images := map[string]string{
+			server:  serverImage,
+			server2: serverImage,
+			server3: helpers.HttpdImage,
+			client2: helpers.NetperfImage,
+		}
+
+		switch mode {
+		case helpers.Create:
+			for k, v := range images {
+				vm.ContainerCreate(k, v, helpers.CiliumDockerNetwork, fmt.Sprintf("-l id.%s", k))
+			}
+			cmd := fmt.Sprintf(
+				"docker run -dt --name netcat --net %s -l id.server-4 busybox sleep 30000s",
+				helpers.CiliumDockerNetwork)
+			vm.Exec(cmd)
+
+			cmd = fmt.Sprintf(
+				"docker run -dt -v %s:/nc.py --net=%s --name client -l id.client python:2.7.14",
+				vm.GetFullPath(ctCleanUpNC), helpers.CiliumDockerNetwork)
+			vm.Exec(cmd)
+
+		case helpers.Delete:
+			for _, x := range containersNames {
+				vm.ContainerRm(x)
+			}
+		}
+	}
+
+	containersMeta := func() map[string]map[string]string {
```

Comment:
Please add comments explaining what these strings represent. Since types aren't being used here, it's important to document what these strings represent.